libs/langchain_v1/langchain/agents/middleware/human_in_the_loop.py

```diff
@@ -1,135 +1,268 @@
+        allow_ignore: Whether the human can choose to ignore/skip the current step
+        allow_response: Whether the human can provide a text response/feedback
+        allow_edit: Whether the human can edit the provided content/state
+    """
+
+    allow_approve: bool
+    allow_ignore: bool
+    allow_response: bool
+    allow_edit: bool
+
+
+class HumanInterrupt(TypedDict):
+    """Represents an interrupt triggered by the graph that requires human intervention.
+
+    This is passed to the `interrupt` function when execution is paused for human input.
+
+    Attributes:
+        action: The action being requested (a tool name)
+        args: Arguments for the action (tool kwargs)
+        config: Configuration defining what actions are allowed
+        description: Optional detailed description of what input is needed
+        tool_call_id: Identifier for the associated tool call
+
+    Example:
+        # Send the interrupt request and get the response
+        request = HumanInterrupt(
+            action="run_command",  # The action being requested
+            args={"command": "ls", "args": ["-l"]},  # Arguments for the action
+            config=HumanInterruptConfig(
+                allow_ignore=True,  # Allow skipping this step
+                allow_response=True,  # Allow text feedback
+                allow_edit=False,  # Don't allow editing
+                allow_approve=True,  # Allow direct acceptance
+            ),
+            description="Please review the command before execution",
+            tool_call_id="call_123",
+        )
+        # Send the interrupt request and get the response
+        response = interrupt([request])[0]
+        ```
+    """
+
+    action: str
+    args: dict
+    config: HumanInterruptConfig
+    description: str | None
+    tool_call_id: str
+
+
+class ApprovePayload(TypedDict):
+    """Human chose to approve the current state without changes."""
+
+    type: Literal["approve"]
+    tool_call_id: str
+
+
+class IgnorePayload(TypedDict):
+    """Human chose to ignore/skip the current step with optional tool message customization."""
+
+    type: Literal["ignore"]
+    tool_call_id: str
+    tool_message: NotRequired[str | ToolMessage]
+
+
+class ResponsePayload(TypedDict):
+    """Human provided text feedback or instructions."""
+
+    type: Literal["response"]
+    tool_call_id: str
+    tool_message: str | ToolMessage
+
+
+class EditPayload(TypedDict):
+    """Human chose to edit/modify the current state/content."""
+
+    type: Literal["edit"]
+    tool_call_id: str
+    action: str
+    args: dict
+
+
+HumanResponse = ApprovePayload | IgnorePayload | ResponsePayload | EditPayload
 
 
 class HumanInTheLoopMiddleware(AgentMiddleware):
     """Human in the loop middleware."""
 
     def __init__(
         self,
-        tool_configs: ToolInterruptConfig,
+        tool_configs: dict[str, bool | HumanInterruptConfig],
```

Comment:
What is the expected behavior for tool calls that haven't been configured via tool_configs? (Could we document what we expect the middleware to do? e.g., raise or auto-approve?)